### PR TITLE
Auto-fuzz: Create dockerfile for fuzzer generator

### PR DESCRIPTION
**Pull request by arthurscchan**
This PR creates a dockerfile for building a custom docker image. The custom docker image will be used for the fuzzer generator for different language to avoid the need to installed specific python package.

### CODE REVIEW

File: tools/auto-fuzz/fuzzer_generator/docker/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+FROM ubuntu:22.04
+
+RUN apt-get update && apt-get install git pip curl -y
+
+RUN curl -sSL https://get.docker.com/ | sh
+RUN git clone https://github.com/ossf/fuzz-introspector fuzz-introspector
```
Comment (DavidKorczynski) [inline tools/auto-fuzz/fuzzer_generator/docker/Dockerfile:20]: Could we use the local copy of fuzz-introspector instead? I assume at least for now we would want to prefer a set up that makes it easy to develop on

**Review by DavidKorczynski — CHANGES_REQUESTED** [review]

File: tools/auto-fuzz/fuzzer_generator/docker/Dockerfile (hunk 1 above)
Comment (arthurscchan) [reply tools/auto-fuzz/fuzzer_generator/docker/Dockerfile]: Changed.

**Review by DavidKorczynski — CHANGES_REQUESTED** [review]
Could you give some explanation on how to use this?

File: tools/auto-fuzz/fuzzer_generator/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+RUN mkdir -p $OUT
+
+RUN apt-get update && apt-get install pip git -y
+
+COPY fuzz-introspector work $SRC/
+
+WORKDIR $SRC/fuzz-introspector
+RUN git submodule init
+RUN git submodule update
+RUN pip3 install -r $SRC/requirements.txt
```
Comment (DavidKorczynski) [inline tools/auto-fuzz/fuzzer_generator/docker/Dockerfile:30]: Could you just do all of the four above commands in a single `RUN` docker command?

**Review by DavidKorczynski — CHANGES_REQUESTED** [review]

File: tools/auto-fuzz/fuzzer_generator/docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+RUN mkdir -p $SRC
+RUN mkdir -p $OUT
+
+RUN apt-get update && apt-get install pip git -y
+
+COPY fuzz-introspector work $SRC/
+
+WORKDIR $SRC/fuzz-introspector
+RUN git submodule init && git submodule update && pip3 install -r ./requirements.txt
```
Comment (DavidKorczynski) [inline tools/auto-fuzz/fuzzer_generator/docker/Dockerfile:28]: please do a line per cmd as oss-fuzz Dockerfiles https://github.com/google/oss-fuzz/blob/master/infra/base-images/base-builder/Dockerfile

**Review by DavidKorczynski — CHANGES_REQUESTED** [review]

**Review by DavidKorczynski — APPROVED** [review]

**Pull request merged**